### ISSUE TITLE
[Product Pull Request] fix: make iframe wrapper take all vieport width

### ISSUE THREAD

**Issue by jmakowski1123** [openedx/platform-roadmap#268]
### For Contributing Author:

This is the Primary Product Ticket for the following community contribution: Add a toggle to allow redirecting to courseware after enrollment

### Checklist prior to undergoing Product Review:

The following information is required in order for Product Managers to be able to review your pull request:

- [x] Explanation of the problem being solved
- [x] Description of how users will be impacted, and which users will be impacted
- [x] Screenshots or video showing the functionality or fix, before and after
- [x] Reproduction steps and/or testing steps

Only if necessary:

- [ ] If necessary, links to corresponding configuration changes
- [ ] If necessary, links to corresponding enablement changes, particularly waffle/toggle status details

### Related PRs

- [ ] https://github.com/openedx/edx-platform/pull/31715

### For Product Manager doing the review:

What criteria should be analyzed from Product to approve a PR?

- The problem being solved by the feature or fix is clear.
- There is clarity on how the change or fix will impact the end user.
- It is clear that the change will not negatively impact users or other areas of the platform.
- The change is implemented comprehensively.
- Any changes to UI use the current, standard Paragon Design System: https://paragon-openedx.netlify.app/

**Comment by jmakowski1123** [comment]
Original PR:

Resolves https://github.com/openedx/xblock-drag-and-drop-v2/issues/312. Probably it was introduced in https://github.com/openedx/frontend-app-learning/pull/27.

Consider how the block interface behaves in the legacy courseware (Lilac):

 2023-04-02.16-46-58.mp4 
On the video you can see that when the viewport reaches 480px, the block resets content width and makes it scrollable. This was implemented this way intentionally in this PR: https://github.com/openedx/xblock-drag-and-drop-v2/pull/135.

Here is how it behaves in this MFE:

 2023-04-02.16-49-45.mp4 
Something weird happens when the viewport width reaches 574px. The iframe shrinks, making XBlock very hard to use.

With changes from this PR:

 2023-04-02.16-54-47.mp4 
It basically removes / compensates padding from the iframe's parent containers and sets 100% width for the iframe container itself. You can see that the block now behaves as in the legacy courseware.

Testing instructions
Switch $(DEVSTACK_ROOT)/frontend-app-learning to this branch.
Verify that xblock-drag-and-drop-v2's content behaves exactly as on the third video in different browsers.
Misc notes
You may notice this bug on the videos:
image

It's not related to this PR's changes, but possibly to this: https://github.com/openedx/frontend-app-learning/pull/414

[private-ref](https://tasks.opencraft.com/browse/BB-7211)

**Comment by jmakowski1123** [comment]
@cablaa77 This one is for product review. For some reason I can't move the screenshots over from the original PR, so you can see them here. https://github.com/openedx/frontend-app-learning/pull/1094

**Comment by itsjeyd** [comment]
Hi @jmakowski1123 @cablaa77 @Daniel-hershel, what can we do to get product review started for this feature?

CC:

@0x29a for https://github.com/openedx/frontend-app-learning/pull/1094
@xitij2000 @mphilbrick211 for https://github.com/openedx/edx-platform/pull/31715

**Comment by ali-hugo** [comment]
@itsjeyd I came across this PR and decided to look into it. 

I noticed something on the final video posted on [1094](https://github.com/openedx/frontend-app-learning/pull/1094). At 0:00, the course navigation fills the whole viewport width:

![0:00](https://github.com/openedx/platform-roadmap/assets/88187719/1247021c-3d25-41cc-ad72-08ffd213fd4f)


But, at 0:29 the navigation is narrower than the viewport width:

![0:29](https://github.com/openedx/platform-roadmap/assets/88187719/6d3fb5a3-0e83-49e3-a885-efc60f551d0b)


Is this expected behaviour? Would it be possible for the navigation to fill the entire wrap?

**Comment by 0x29a** [comment]
@ali-hugo, this is not expected, but I think it's a separate bug. I mentioned it in the "Misc notes" section of the PR, but perhaps should have added more details:
![image](https://github.com/openedx/platform-roadmap/assets/18251194/eeb7e9a6-56f3-4089-8fca-f5b80ad16e1c)

**Comment by itsjeyd** [comment]
> I came across this PR and decided to look into it.

Thanks a lot @ali-hugo!

**Comment by ali-hugo** [comment]
> @ali-hugo, this is not expected, but I think it's a separate bug. I mentioned it in the "Misc notes" section of the PR, but perhaps should have added more details:

@0x29a Oh, I see. I didn't realise the icon was supposed to appear next to the navigation. Got it! Thanks for the diagram.

**Comment by antoviaque** [comment]
@ali-hugo Thank you for the review here! :+1: 

@jmakowski1123 Will @ali-hugo 's review be binding, ie will be sufficient to complete the product review, and allow the PR to merge once the issues fixed? Or will it still require a product review?

**Comment by itsjeyd** [comment]
(Side note for additional context: https://github.com/openedx/edx-platform/pull/31715 hasn't gone through engineering review yet. So regardless of who gets to provide final/binding approval from product perspective, there will be one more step to complete before we can merge it.)

**Comment by itsjeyd** [comment]
@ali-hugo Is this good from your side now and if so, would you mind leaving an explicit approval?

Depending on @jmakowski1123's input on @antoviaque's question above, we might still need another 👍 from product, but it would be good to know if you have any additional feedback for @0x29a to address.

**Comment by itsjeyd** [comment]
@jmakowski1123 I just noticed that the info about related PRs in the issue description might need updating? In particular, the following parts seem to be referencing a different feature and PR than https://github.com/openedx/frontend-app-learning/pull/1094:

> ### For Contributing Author:
> 
> This is the Primary Product Ticket for the following community contribution: Add a toggle to allow redirecting to courseware after enrollment
> Checklist prior to undergoing Product Review:
> 
> ...
> 
> ### Related PRs
> 
> https://github.com/openedx/edx-platform/pull/31715

**Comment by ali-hugo** [comment]
> @ali-hugo Is this good from your side now and if so, would you mind leaving an explicit approval?

@itsjeyd @0x29a 

 I took another look at the videos on [#1094](https://github.com/openedx/frontend-app-learning/pull/1094) and everything looks good to me. You can consider this my official approval! ✅

**Comment by itsjeyd** [comment]
Great, thanks for the quick reply @ali-hugo 🎉

**Comment by itsjeyd** [comment]
Hey @jmakowski1123, when you get a minute would you mind having a look at these questions:

* https://github.com/openedx/platform-roadmap/issues/268#issuecomment-1621910825
* https://github.com/openedx/platform-roadmap/issues/268#issuecomment-1631048782

**Comment by Agrendalath** [comment]
@itsjeyd, the PR is merged now. Should we change the label and close this issue?

**Comment by itsjeyd** [comment]
@Agrendalath Updating feature tickets on the [Open edX Roadmap](https://github.com/orgs/openedx/projects/4) board is currently out-of-scope for OSPR management, so I'll let @jmakowski1123 handle this part.

@jmakowski1123 Happy to ping someone else from the product team if that's what you'd prefer; just let me know.

(In the meantime I did change the label from `ready for product review` to `product review done` to avoid unnecessary churn for product reviewers looking for PRs to review.)

**Comment by sarina** [comment]
Closing as this is marked "Shipped" on the PR board 🎉
